aws-xray-auto-instrumentation-agent-runtime/src/main/java/com/amazonaws/xray/agent/config/AgentConfiguration.java

```diff
@@ -0,0 +1,194 @@
+        tracingEnabled = true;
+    }
+
+    /**
+     * Constructs agent configuration from provided map. Does not do any validation since that is taken care of in
+     * {@link XRaySDKConfiguration}. Can throw {@code InvalidAgentConfigException} if improper type is assigned to any
+     * property. Assigns all non-configured properties to default values.
+     * @param properties - Map of property names to their values in string representation, which will be cast to proper types.
+     */
+    public AgentConfiguration(Map<String, String> properties) {
+        String serviceName = DEFAULT_SERVICE_NAME,
+                contextMissingStrategy = "LOG_ERROR",
+                daemonAddress = "127.0.0.1:2000",
+                samplingStrategy = "CENTRAL",
+                samplingRulesManifest = null,
+                awsServiceHandlerManifest = null;
+        int maxStackTraceLength = 50,
+                streamingThreshold = 100,
+                awsSDKVersion = 2;
+        boolean tracingEnabled = true;
+
+        if (properties != null) {
+            try {
+                for (Map.Entry<String, String> entry : properties.entrySet()) {
+                    switch (entry.getKey()) {
+                        case "serviceName":
+                            serviceName = entry.getValue();
+                            break;
+                        case "contextMissingStrategy":
+                            contextMissingStrategy = entry.getValue();
+                            break;
+                        case "daemonAddress":
+                            daemonAddress = entry.getValue();
+                            break;
+                        case "samplingStrategy":
+                            samplingStrategy = entry.getValue();
+                            break;
+                        case "maxStackTraceLength":
+                            maxStackTraceLength = Integer.parseInt(entry.getValue());
+                            break;
+                        case "streamingThreshold":
+                            streamingThreshold = Integer.parseInt(entry.getValue());
+                            break;
+                        case "samplingRulesManifest":
+                            samplingRulesManifest = entry.getValue();
+                            break;
+                        case "awsSDKVersion":
+                            awsSDKVersion = Integer.parseInt(entry.getValue());
+                            break;
+                        case "awsServiceHandlerManifest":
+                            awsServiceHandlerManifest = entry.getValue();
+                            break;
+                        case "tracingEnabled":
+                            tracingEnabled = Boolean.parseBoolean(entry.getValue());
+                            break;
+                        default:
+                            log.debug("Encountered unknown property " + entry.getKey() + " in X-Ray agent configuration. Ignoring.");
```

Comment:
My previous point probably adds some context, `debug` seems too low for this. Not sure about info vs warn but lean towards warn.

Reply:
Good point, I'll change to `warn`. Also seems like I missed a couple comments from last review, my bad!